pyproj/_crs.pyx

```diff
@@ -215,97 +473,439 @@ cdef class Ellipsoid:
+
+
+cdef class CoordinateOperation(Base):
+    """
+    Coordinate operation for CRS.
+
+    Attributes
+    ----------
+    name: str
+        The name of the method(projection) with authority information.
+    method_name: str
+        The method (projection) name.
+    method_auth_name: str
+        The method authority name.
+    method_code: str
+        The method code.
+    is_instantiable: int
+        If 1, a coordinate operation can be instantiated as a PROJ pipeline.
+        This also checks that referenced grids are available. 
+    has_ballpark_transformation: int
+        If 1, the coordinate operation has a “ballpark” transformation, 
+        that is a very approximate one, due to lack of more accurate transformations. 
+    accuracy: float
+        The accuracy (in metre) of a coordinate operation. 
+
+    """
+    def __cinit__(self):
+        self._params = None
+        self._grids = None
+        self.method_name = "undefined"
+        self.method_auth_name = "undefined"
+        self.method_code = "undefined"
+        self.is_instantiable = False
+        self.has_ballpark_transformation = False
+        self.accuracy = float("nan")
+
+    @staticmethod
+    cdef create(PJ_CONTEXT* projcontext, PJ* projobj):
+        cdef CoordinateOperation coord_operation = CoordinateOperation()
+        coord_operation.projobj = proj_crs_get_coordoperation(
+            projcontext,
+            projobj
+        )
+        if coord_operation.projobj == NULL:
+            return None
+
+        cdef char *out_method_name = NULL
+        cdef char *out_method_auth_name = NULL
+        cdef char *out_method_code = NULL
+        proj_coordoperation_get_method_info(
+            projcontext,
+            coord_operation.projobj,
+            &out_method_name,
+            &out_method_auth_name,
+            &out_method_code
+        )
+        coord_operation.method_name = decode_or_undefined(out_method_name)
+        coord_operation.method_auth_name = decode_or_undefined(out_method_auth_name)
+        coord_operation.method_code = decode_or_undefined(out_method_code)
+        coord_operation.name = "{method_name} [{method_auth_name}:{method_code}]".format(
+            method_name=coord_operation.method_name,
+            method_auth_name=coord_operation.method_auth_name,
+            method_code=coord_operation.method_code,
+        )
+        coord_operation.accuracy = proj_coordoperation_get_accuracy(
+            projcontext,
+            coord_operation.projobj
+        )
+        coord_operation.is_instantiable = proj_coordoperation_is_instantiable(
+            projcontext,
+            coord_operation.projobj
+        )
+        coord_operation.has_ballpark_transformation = \
+            proj_coordoperation_has_ballpark_transformation(
+                projcontext,
+                coord_operation.projobj
+            )
+
+        # TODO: How do you get the value count?
+        # The example in the PROJ test just says 7 ....
```

Comment:
By definition it are always 3 or 7 parameters I think

Reply:
My main question is how do we know? Maybe a question for the PROJ maintainers.

Reply:
https://github.com/OSGeo/proj.4/issues/1430